backend/generalhw.pm

```diff
@@ -52,13 +53,18 @@ sub run_cmd ($self, @args) {
     my @full_cmd = split / /, $self->get_cmd($args[0]);
 
     my ($stdin, $stdout, $stderr, $ret);
-    eval { $ret = IPC::Run::run([@full_cmd], \$stdin, \$stdout, \$stderr) };
-    die "Unable to run command '@full_cmd' (deduced from test variable $args[0]): $@\n" if $@;
+
+    {
+        # Do not let the SIGCHLD handler of Mojo::IOLoop::ReadWriteProcess::Session steal the exit code from IPC::Run
+        local $SIG{CHLD};
+        eval { $ret = IPC::Run::run([@full_cmd], \$stdin, \$stdout, \$stderr) };
+        die "Unable to run command '@full_cmd' (deduced from test variable $cmd): $@\n" if $@;
+    }
     chomp $stdout;
     chomp $stderr;
 
-    die "$args[0]: $stderr" unless $ret;
-    bmwqemu::diag("IPMI: $stdout");
+    die "$cmd: stdout: $stdout, stderr: $stderr" unless $ret;
```

Comment:
`$cmd` is only introduced in https://github.com/os-autoinst/os-autoinst/pull/1898/commits/e2fc32e2b2743ada8e27e1b87f9d890f63e836fc

Reply:
ok, removed the logging change commit

Reply:
added it back due to https://github.com/os-autoinst/os-autoinst/pull/1918#issuecomment-1013909982